clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -660,8 +657,9 @@ public int write(ByteBuffer src) throws IOException {
             throw renegotiationException();
 
         if (wrapResult.getStatus() == Status.OK) {
-            written = wrapResult.bytesConsumed();
+            int written = wrapResult.bytesConsumed();
```

Comment:
At this point, it's not clear if all bytes from src have been read. So, it might be useful to loop through the writes to sslEngine until flush() can't write all bytes to sockets and then return. This can potentially avoid the caller from re-populating the same bytes from fileChannel again.

Reply:
I updated the code to loop until either the socket buffer is full (i.e. `flush` returns `false`) or there are no remaining bytes in `src`. Please take a look and see if this is what you had in mind.